apps/src/templates/projects/ProjectAppTypeArea.jsx

```diff
@@ -105,12 +112,42 @@ const ProjectAppTypeArea = React.createClass({
     const {hasOlderProjects} = this.props;
     if (this.state.maxNumProjects < newNumProjects && hasOlderProjects) {
       this.setState({disableViewMore: true});
-      this.props.fetchOlderProjects(this.props.labKey, () => {
+      this.fetchOlderProjects(() => {
         this.setState({disableViewMore: false});
       });
     }
   },
 
+  /**
+   * Fetch additional projects of the specified type which were published
+   * earlier than the oldest published project currently in our list.
+   * @param {function} callback Callback to call after network request completes.
+   */
+  fetchOlderProjects(callback) {
+    const {projectList, labKey: projectType} = this.props;
+    const oldestProject = projectList[projectList.length - 1];
+    const oldestPublishedAt = oldestProject && oldestProject.projectData.publishedAt;
+
+    $.ajax({
+      method: 'GET',
+      url: `/api/v1/projects/gallery/public/${projectType}/${MAX_PROJECTS_PER_CATEGORY}/${oldestPublishedAt}`,
+      dataType: 'json'
+    }).done(data => {
+      // olderProjects all have an older publishedAt date than oldestProject.
+      const olderProjects = data[projectType];
+
+      // Don't try to fetch projects of this projectType again in the future if we
+      // received fewer than we asked for this time.
+      if (olderProjects.length < MAX_PROJECTS_PER_CATEGORY) {
+        this.props.setHasOlderProjects(false, projectType);
+      }
+
+      // Append any projects we just received to the appropriate list,
+      // ignoring any duplicates.
+      this.props.addOlderProjects(olderProjects, projectType);
```

Comment:
I wonder if this should just be something like `appendProjects`? I found myself confused by these two uses of olderProjects, where (according to my understanding) `setHasOlderProjects` is saying whether the server has projects left that are older than what's on the client right now, and `addOlderProjects` is saying take this set of projects and incorporate them on your list so that they become part of your current projects.